hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/clustering/ClusteringPlanOperator.java

```diff
@@ -88,10 +97,20 @@ public void notifyCheckpointComplete(long checkpointId) {
   }
 
   private void scheduleClustering(HoodieFlinkTable<?> table, long checkpointId) {
+    List<HoodieInstant> pendingClusteringInstantTimes =
+        ClusteringUtils.getPendingClusteringInstantTimes(table.getMetaClient());
     // the first instant takes the highest priority.
     Option<HoodieInstant> firstRequested = Option.fromJavaOptional(
-        ClusteringUtils.getPendingClusteringInstantTimes(table.getMetaClient()).stream()
+        pendingClusteringInstantTimes.stream()
             .filter(instant -> instant.getState() == HoodieInstant.State.REQUESTED).findFirst());
+
+    long pendingClusteringCount = pendingClusteringInstantTimes.stream()
+        .filter(instant -> instant.getState() == State.REQUESTED).collect(Collectors.toList()).size();
```

Comment:
Can we use `count` of stream instead of collect ?

Reply:
ok